### PR TITLE
✨ Source BambooHR - Add Employee Directory and Time Off Request Streams

### PR DESCRIPTION
**Pull request by KimPlv**
## What
Add a new streams in Source BambooHR 

## How
- [Employee Directory](https://documentation.bamboohr.com/reference/get-employees-directory-1) full-refresh only

- [Time Off Requests](https://documentation.bamboohr.com/reference/time-off-get-time-off-requests-1) full-refresh and incremental 
need to do a request loop with interval days (fix to 60) to get all time off requests with status updated. 
API don't allow to incremental on created or lastChanged but on start (Only show time off that occurs on/after the specified start date) and end (Only show time off that occurs on/before the specified end date)

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- Connector version is set to `0.1.0`
    - `Dockerfile` has version `0.1.0`
- Documentation updated
    - `docs/integrations/source/bamboo-hr.md` including changelog with an entry for the initial version. 
    - `docs/integrations/README.md`

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "$schema": "https://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "additionalProperties": false,
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:4]: It's better define as true to be easier to manage in future updates.

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "properties": {
+        "id": {
+            "type": "string"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:7]:
```suggestion
            "type": ["null", "string"]
```

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        "name": {
+            "type": "string"
+        },
+        "start": {
+            "type": "string",
+            "default": ""
+        },
+        "end": {
+            "type": "string",
+            "default": ""
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:21]:
```suggestion
            "type": ["null", "string"],
```

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/spec.json [hunk 4]
```diff
@@ -16,6 +19,15 @@
         "description": "Api key of bamboo hr",
         "airbyte_secret": true
       },
+      "start_date": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/spec.json:22]: Adding a new parameter is a breaking change to existing connections.

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/utils.py [hunk 5]
```diff
@@ -15,3 +16,16 @@ def validate_custom_fields(custom_fields, available_fields):
             denied_fields.append(custom_field)
 
     return denied_fields
+
+
+def daterange(date1, date2):
+        for n in range(int((date2 - date1).days)+1):
+            yield date1 + timedelta(n)
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/utils.py:23]: Please format the function and add unit tests for all of them.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@KimPlv thanks for the contribution. At the moment the team is working to migrate BambooHR to lowcode framework and I'm not sure if adding this new endpoint with 2 custom functions is the best time.

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/spec.json (hunk 4 above)
Comment (KimPlv) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/spec.json:22]: Yes but we need this parameter to do incremental

**Comment by marcosmarxm** [comment]
@KimPlv the low-code migrated was merged. Do you mind submitting a new pull request adding the new streams to the new format? I'll close this one as it is doing changes to an old format.

**Pull request closed without merging**